### PR TITLE
[OSAC-374] - WIP - Invoke Powershell - crossplatform

### PR DESCRIPTION
**Pull request by nicolae-anghel**
https://uipath.atlassian.net/browse/OSAC-374

### CODE REVIEW

File: Activities/.nuget/NuGet.targets [hunk 1]
```diff
@@ -27,12 +27,12 @@
 
     <PropertyGroup Condition=" '$(OS)' == 'Windows_NT'">
         <!-- Windows specific commands -->
-        <NuGetToolsPath>$([System.IO.Path]::Combine("$(ProjectDir)..\..\", ".nuget"))</NuGetToolsPath>
+        <NuGetToolsPath>$([System.IO.Path]::Combine($(SolutionDir), ".nuget"))</NuGetToolsPath>
```
Comment (alcxs) [inline Activities/.nuget/NuGet.targets:30]: this needs to be validated against the other APs

File: Activities/.nuget/NuGet.targets [hunk 2]
```diff
@@ -27,12 +27,12 @@
...
     </PropertyGroup>
 
     <PropertyGroup Condition=" '$(OS)' != 'Windows_NT'">
         <!-- We need to launch nuget.exe with the mono command if we're not on windows -->
-        <NuGetToolsPath>$(ProjectDir)..\..\.nuget</NuGetToolsPath>
+        <NuGetToolsPath>$(SolutionDir).nuget</NuGetToolsPath>
```
Comment (alcxs) [inline Activities/.nuget/NuGet.targets:35]: same as above

File: Activities/Directory.build.targets [hunk 3]
```diff
@@ -1,23 +1,30 @@
...
-	<PackageReference Update="UiPath.Activities.Api" Version="22.4.1" PrivateAssets="All" />
-	<PackageReference Update="UiPath.Activities.Contracts" Version="22.4.1" PrivateAssets="All" />
-	<PackageReference Update="UiPath.Telemetry.Client" Version="1.3.18" PrivateAssets="All" />
-	<PackageReference Update="UiPath.Platform" Version="22.4.1"  PrivateAssets="All" />
-  </ItemGroup>
-  <PropertyGroup>
-    <DefineConstants Condition="'$(TargetFramework)' == 'net5.0'">$(DefineConstants);NETSTANDARD</DefineConstants>
-  </PropertyGroup>
+	<ItemGroup Condition="$(TargetFramework.StartsWith('net6'))">
+		<PackageReference Update="System.Activities" Version="6.0.0-20220106-04" />
```
Comment (alcxs) [inline Activities/Directory.build.targets:3]: Update this to `UiPath.Workflow.Runtime` and match versions as in this document: https://docs.uipath.com/developer/docs/migrating-activities-to-net-5#:~:text=%3CPackageReference%20Update%3D%22UiPath.Workflow.Runtime%22%20Version%3D%226.0.0%2D20220401%2D03%22%20PrivateAssets%3D%22All%22/%3E

File: Activities/Scripting/UiPath.Scripting.Activities.Design/UiPath.Scripting.Activities.Design.csproj [hunk 4]
```diff
@@ -0,0 +1,60 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop" ToolsVersion="Current">
+	<Import Project="..\Scripting.build.props" />
+	<PropertyGroup>
+		<TargetFrameworks>net6.0-windows</TargetFrameworks>
+		<OutputPath>$(ProjectDir)..\..\Output\Activities\Scripting\</OutputPath>
+		<UseWPF>true</UseWPF>
+	</PropertyGroup>
+	<ItemGroup Condition=" '$(TargetFramework)' == 'net6.0-windows' ">
+		<PackageReference Include="System.Activities" />
```
Comment (alcxs) [inline Activities/Scripting/UiPath.Scripting.Activities.Design/UiPath.Scripting.Activities.Design.csproj:9]: same as above, modify to "UiPath.Workflow.Runtime"

File: Activities/Scripting/UiPath.Scripting.Activities.Design/UiPath.Scripting.Activities.Design.csproj [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		<ProjectReference Include="..\UiPath.Scripting\UiPath.Scripting.csproj">
+			<Project>{76BB5A59-B73C-4643-9A6F-E8F57DB6CABD}</Project>
+			<Name>UiPath.Scripting</Name>
+		</ProjectReference>
+	</ItemGroup>
+	<ItemGroup>
+		<None Include="UiPath.Scripting.Activities.Design.nuspec" />
+	</ItemGroup>
+	<ItemGroup>
+		<PackageReference Include="UiPath.Activities.Api" Version="21.10.4" />
```
Comment (alcxs) [inline Activities/Scripting/UiPath.Scripting.Activities.Design/UiPath.Scripting.Activities.Design.csproj:52]: remove version from here, leave it only in build.targets file

File: Activities/Scripting/UiPath.Scripting.Activities.Packaging/UiPath.Scripting.Activities.Packaging.csproj [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+		<VersionSuffix></VersionSuffix>
+		<GeneratePackageOnBuild Condition="'$(Configuration)' == 'Release'">false</GeneratePackageOnBuild>
+		<GeneratePackageOnBuild Condition="'$(Configuration)' == 'Debug'">true</GeneratePackageOnBuild>
+		<PackageOutputPath Condition="'$(Configuration)' == 'Debug'">..\..\Output\Activities\Packages\</PackageOutputPath>
+		<AllowedOutputExtensionsInPackageBuildOutputFolder>$(AllowedOutputExtensionsInPackageBuildOutputFolder);.pdb</AllowedOutputExtensionsInPackageBuildOutputFolder>
+	</PropertyGroup>
+
+	<PropertyGroup>
+		<PackageId>Nick.UiPath.Scripting.Activities</PackageId>
+		<PackageTitle>Nick.UiPath.Scripting.Activities</PackageTitle>
```
Comment (alcxs) [inline Activities/Scripting/UiPath.Scripting.Activities.Packaging/UiPath.Scripting.Activities.Packaging.csproj]: please cleanup this :)

File: Activities/Scripting/UiPath.Scripting.Activities.Packaging/UiPath.Scripting.Activities.Packaging.csproj [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+		<PackageIconUrl>https://download.uipath.com/UI_icon.png?web</PackageIconUrl>
+		<PackageLicenseUrl>https://docs.uipath.com/activities/docs/about-the-scripting-activities-pack</PackageLicenseUrl>
+		<PackageTags>UiPathActivities</PackageTags>
```
Comment (alcxs) [inline Activities/Scripting/UiPath.Scripting.Activities.Packaging/UiPath.Scripting.Activities.Packaging.csproj:19]: check with PM what should be the actual tags here

File: Activities/Scripting/UiPath.Scripting.Activities.Packaging/UiPath.Scripting.Activities.Packaging.csproj [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+	</PropertyGroup>
+
+	<Target Name="RemoveMetaDll" AfterTargets="BuiltProjectOutputGroup">
+		<ItemGroup>
+			<BuiltProjectOutputGroupOutput Remove="@(BuiltProjectOutputGroupOutput)" />
+		</ItemGroup>
+	</Target>
+
+	<ItemGroup Condition=" '$(TargetFramework)' == 'net6.0' ">
+		<PackageReference Include="System.Activities" PrivateAssets="All"/>
```
Comment (alcxs) [inline Activities/Scripting/UiPath.Scripting.Activities.Packaging/UiPath.Scripting.Activities.Packaging.csproj:34]: same as above

File: Activities/Scripting/UiPath.Scripting.Activities/UiPath.Scripting.Activities.csproj [hunk 9]
```diff
@@ -0,0 +1,40 @@
+﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="Current">
+	<Import Project="..\Scripting.build.props" />
+	<PropertyGroup>
+		<PackageId>.noconflict</PackageId>
+		<TargetFrameworks>net6.0</TargetFrameworks>
+		<OutputPath>..\..\Output\Activities\Scripting\</OutputPath>
+	</PropertyGroup>
+	<ItemGroup Condition=" '$(TargetFramework)' == 'net6.0' ">
+		<PackageReference Include="System.Activities" />
```
Comment (alcxs) [inline Activities/Scripting/UiPath.Scripting.Activities/UiPath.Scripting.Activities.csproj:9]: same as above - replace with `UiPath.Workflow.Runtime`

**Review by alcxs — CHANGES_REQUESTED** [review]
see individual comments

File: Activities/Scripting/UiPath.Scripting.Activities.Packaging/UiPath.Scripting.Activities.Packaging.csproj (hunk 6 above)
Comment (nicolae-anghel) [reply Activities/Scripting/UiPath.Scripting.Activities.Packaging/UiPath.Scripting.Activities.Packaging.csproj]: Fixed

**Comment by alcxs** [comment]
Closing this as a new one was opened https://github.com/UiPath/Community.Activities/pull/278

**Pull request closed without merging**